packages/web/app/utils/survey.jsx

```diff
@@ -46,15 +52,21 @@ const QUESTION_COMPONENTS = {
   [PROGRAM_DATA_ELEMENT_TYPES.SELECT]: BaseSelectField,
   [PROGRAM_DATA_ELEMENT_TYPES.MULTI_SELECT]: BaseMultiselectField,
   [PROGRAM_DATA_ELEMENT_TYPES.AUTOCOMPLETE]: SurveyQuestionAutocompleteField,
-  [PROGRAM_DATA_ELEMENT_TYPES.DATE]: (props) => <DateField {...props} saveDateAsString />,
-  [PROGRAM_DATA_ELEMENT_TYPES.DATE_TIME]: (props) => <DateTimeField {...props} saveDateAsString />,
-  [PROGRAM_DATA_ELEMENT_TYPES.SUBMISSION_DATE]: (props) => (
-    <DateField {...props} saveDateAsString />
-  ),
+  [PROGRAM_DATA_ELEMENT_TYPES.DATE]: props => <DateField {...props} saveDateAsString />,
+  [PROGRAM_DATA_ELEMENT_TYPES.DATE_TIME]: props => <DateTimeField {...props} saveDateAsString />,
+  [PROGRAM_DATA_ELEMENT_TYPES.SUBMISSION_DATE]: props => <DateField {...props} saveDateAsString />,
   [PROGRAM_DATA_ELEMENT_TYPES.NUMBER]: NumberField,
   [PROGRAM_DATA_ELEMENT_TYPES.BINARY]: NullableBooleanField,
   [PROGRAM_DATA_ELEMENT_TYPES.CHECKBOX]: NullableBooleanField,
-  [PROGRAM_DATA_ELEMENT_TYPES.CALCULATED]: ReadOnlyTextField,
+  [PROGRAM_DATA_ELEMENT_TYPES.CALCULATED]: ({ field, ...props }) => (
+    <ReadOnlyTextField
+      {...props}
+      field={{
+        ...field,
+        value: getStringOfCalculatedValue(field?.value),
```

Comment:
A change in behaviour with this is that now if the value isn't a number, we'll hit an uncaught error here. Since that's a mis-configuration and the user would be hitting an error on submission previously I think that's okay. But arguably a toast message would be cleaner...